events/static/js/app/companies.js

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import axios from 'axios'
+import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
+import getMuiTheme from 'material-ui/styles/getMuiTheme';
+import Paper from 'material-ui/Paper';
+import Chip from 'material-ui/Chip';
+import Avatar from 'material-ui/Avatar';
+import AddCompaniesWindow from './AddCompanyWindow';
+import {List, ListItem} from 'material-ui/List';
+import { hashHistory } from 'react-router'
+import SearchField from './SearchField';
+
+
+export default class CompaniesList extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            companies: [],
+            searchCompanies: [],
+
+        };
+
+        this.loadCompanies = this.loadCompanies.bind(this);
+        this.handleAdminClick = this.handleAdminClick.bind(this);
+        this.newDataHandler = this.newDataHandler.bind(this);
+
+    }
+
+    handleAdminClick(event) {
+        event.stopPropagation();
+    }
+
+    newDataHandler(company_id) {
+        hashHistory.push('/companies/'+company_id)
+    }
+
+
+    componentDidMount () {
+        this.loadCompanies()
+    }
+
+    loadCompanies() {
+        axios.get('api/v1/companies')
+            .then((response) => {
+                this.setState({
+                    companies: response.data.companies,
+                    searchCompanies: response.data.companies,
+                });
+            })
+    }
+
+    handleCompanyClick(id) {
+        setTimeout(hashHistory.push('/companies/' + id), 20000);
+    }
+
+    render() {
+
+        let companies = this.state.searchCompanies.map( (companyObject) => {
+            return (
+                <ListItem
+                    onTouchTap={this.handleCompanyClick.bind(this, companyObject['id'])}
+                    key={companyObject['id']}
+                    primaryText={companyObject['name']}
+                >
+                    <div className="outer-div">
+                        <div className="inner-div">
+                            <Chip
+                                onTouchTap={this.handleAdminClick}
+                                style={{
+                                    float: 'right',
+                                    margin: '-8px 0'
+                                }}
+                            >
+                                    <Avatar size={32}>
+                                        {companyObject['admin']['first_name'][0].toUpperCase()}
+                                    </Avatar>
+                                    {
+                                        companyObject['admin']['first_name'] + ' ' +
```

Comment:
I would recommend adding full name method rather than always adding the first and last name.